src/libs/SessionUtils.js

```diff
@@ -0,0 +1,34 @@
+import lodashGet from 'lodash/get';
+
+/**
+ * Determine if the transitioning user is logging in as a new user.
+ *
+ * @param {String} transitionURL
+ * @param {String} sessionEmail
+ * @returns {Boolean}
+ */
+function isLoggingInAsNewUser(transitionURL, sessionEmail) {
+    // The OldDot mobile app does not URL encode the parameters, but OldDot web
+    // does. We don't want to deploy OldDot mobile again, so as a work around we
+    // compare the session email to both the decoded and raw email from the transition link.
+    const params = new URLSearchParams(transitionURL);
+    const paramsEmail = params.get('email');
+
+    // If the email param matches what is stored in the session then we are
+    // definitely not logging in as a new user
+    if (paramsEmail === sessionEmail) {
```

Comment:
Can you confirm this handles all four cases?

1. Unencoded new user (OldApp new user)
2. Encoded new user (OldDot new user)
3. Unencoded existing user (OldApp existing user)
4. Encoded existing user (OldDot existing user)

Specifically interested in case 2? e.g. when do we return true when `paramsEmail !== sessionEmail`?

Reply:
1. Mobile un-encoded as the same user
The params and session emails don't match, but then the linked email matches the session email so we correctly
return `false`.

2. Mobile un-encoded as a different user
The params and session emails don't match, then the linked and session emails also don't match so we correctly return `true`.

3. Web as the same user
The params and session emails match so we correctly return `false`.

4. Web as a different user
The params and session emails don't match so we check the linked and session emails which
also don't match so we correctly return `true`.